content/blogs/2023-10-09-why-ingestion-will-never-be-solved.md

```diff
@@ -0,0 +1,155 @@
+::
+
+![airbyte_kestra.gif](/blogs/2023-10-09-why-ingestion-will-never-be-solved/airbyte_kestra.gif)
+
+Another distinction compared to [Singer](https://www.singer.io/) is that Airbyte is primarily designed to be used from the UI, not via code. This brings us to [CloudQuery](https://www.cloudquery.io/) and [dltHub](https://dlthub.com/) which are code-centric and intended to be used by developers.
+
+### CloudQuery
+
+[CloudQuery](https://github.com/cloudquery/cloudquery) is an open-source ELT framework for developers. CloudQuery syncs can be executed from the terminal independently of the platform. You can maintain source and destination configuration in individual YAML files, and run those connectors locally, on a remote server, Kubernetes, serverless function, or from [kestra](https://demo.kestra.io/ui/blueprints/community?q=cloudquery&page=1).
+
+We'll discuss CloudQuery in more detail in another blog post. For now, make sure to check out our [CloudQuery plugin](https://github.com/kestra-io/plugin-cloudquery) and [blueprints](https://demo.kestra.io/ui/blueprints/community?q=cloudquery&page=1).
+
+How does CloudQuery monetize? CloudQuery offers paid connectors for more advanced integrations, and there's also a plan for a [managed service](https://www.cloudquery.io/register-for-cloud) for running CloudQuery connectors in the cloud.
+
+
+### dltHub
+
+[dltHub](https://github.com/dlt-hub/dlt) is a fairly new open-source data integration library. Note that it's not a platform, but a library that you can use to build your own platform. This offers an unmatched level of flexibility, but it also requires a little more work to figure out the right deployment patterns. As a Python library, you can use dlt anywhere you can use Python. You can run dlt integration jobs from your laptop, remote server, cloud function, or from [kestra](https://demo.kestra.io/ui/blueprints/community?q=dlt&page=1). dltHub connectors are distributed as extensions to the main dlt library.
+
+One of the main selling points of dlt is that it's a lightweight Python-based framework, allowing you to create new connectors with automatic schema inference and schema evolution. You can transform your data along the way e.g. to add pseudomization step before loading data to the target destination. And there are custom decorators to configure incremental and merge loads in Python.
+
+How does dltHub monetize? dlt plans to offer an [integration hub](https://www.cloudquery.io/register-for-cloud) which might resemble a connector marketplace and a managed service for data integration. It seems similar to CloudQuery's offering, but it's too early to tell how the two products will compare commercially.
```

Comment:
wrong link

Reply:
I switched them, need a link for both thanks